docs/user-guides/class_notebooks_best_practices.md

```diff
@@ -1,97 +1,134 @@
-[Return to Table of Contents](../user.md)
-
-# Developing Notebooks for Classes or Trainings: Best Practices
+# Notebook Development Best Practices
+<br>
 
 ## Provide a Conda Environment Capable of Running the Notebooks
 
-- Provide students with a conda environment that has everything they need. [Conda Environments in OpenSARLab.](conda_environments.md)
+- Include an `environment.yml` in your GitHub repository that can be used to build a Conda environment that supports the notebook.
+- Learn how to [Install software in a lab](software.md).
+
+---
+
+## Provide a Pinned `environment.yml` to Build a Conda Environment
 
-- Students can mimic instructor's environment by:
+The open-source Python packages used in your Conda environments are routinely updated, often introducing breaking dependencies. If the `environment.yml` file does not pin packages to specific versions, the default behavior is to build the latest available versions. This can introduce new dependency conflicts at any time. 
 
-    1. Distributing ```environment.yml``` file
-    1. Upload ```environment.yml``` into following directory: ```/home/jovyan/conda_environments/Environment_Configs/```         
-    1. Create using ```Create_OSL_Conda_Environments.ipynb``` notebook located in ```/home/jovyan/conda_environments```
+Even if you pin every package, there is still a risk that an `environment.yml` will no longer build correctly at some point. This can occur if one of its package versions is removed due to a security risk, legal issue, or the discovery of a critical error, though this is relatively rare.
 
-- You may encounter dependency conflicts that can prevent you from installing essential software to run all of your notebooks in a single environment. In such cases, create multiple conda environments to run different notebooks.
+### To reduce the chances of your `environment.yml` breaking over time, provide a pinned version to your users:
+1. Open a terminal.
+2. Create the conda environment if it does not yet exist.
+3. Run the command `mamba env export -n your_environement_name > environement.yml`
+4. Open the newly created `environment.yml` in a text editor.
+5. Remove the `prefix` line at the bottom of the file
+    - `prefix` is the path to the environment on your system, which will likely differ between users.
+6. Save the file.
+7. Push the file to your notebook repository.  
 
 ---
 
-## Set the Notebook Metadata to Use the Correct Environment
+## Set The Notebook Metadata to Use the Correct Environment
```

Comment:
Weirdly inconsistent capitalization, even within the "the"-s

Reply:
Good catch.